src/formatter/apply.ts

```diff
@@ -0,0 +1,21 @@
+import { pipe } from "fp-ts/lib/function";
+import * as OP from "fp-ts/lib/Option";
+import { MappingFormatter } from "./types";
+
+export const applyFormat = <T, O>(
+  dataFlow: T,
+  inputFieldName: keyof T,
+  outputFieldName?: string
+) => (mappingFn: MappingFormatter<T[keyof T], O>): T =>
+  pipe(
+    dataFlow[inputFieldName],
+    mappingFn,
+    output =>
+      pipe(
+        outputFieldName,
+        OP.fromNullable,
+        OP.getOrElseW(() => inputFieldName),
+        fieldName => ({ [fieldName]: output })
+      ),
+    formatted => ({ ...dataFlow, ...formatted })
```

Comment:
If 'outputFieldName' is specified, the old value with the original key remains in the object, resulting in having two keys pointing to different values. Is this intentional, or should we consider removing the old value for consistency?

Reply:
Nope `formatted` should have the original fieldName. In this case spread operation should overwrite the value by keeping the original field name.